scripts/translation/libqa/ArgvParser.php

```diff
@@ -0,0 +1,83 @@
+<?php /*
++----------------------------------------------------------------------+
+| Copyright (c) 1997-2025 The PHP Group                                |
++----------------------------------------------------------------------+
+| This source file is subject to version 3.01 of the PHP license,      |
+| that is bundled with this package in the file LICENSE, and is        |
+| available through the world-wide-web at the following url:           |
+| https://www.php.net/license/3_01.txt.                                |
+| If you did not receive a copy of the PHP license and are unable to   |
+| obtain it through the world-wide-web, please send a note to          |
+| license@php.net, so we can mail you a copy immediately.              |
++----------------------------------------------------------------------+
+| Authors:     André L F S Bacci <ae php.net>                          |
++----------------------------------------------------------------------+
+
+# Description
+
+This class coordinates and centrailzes control for $argv command line
+parameters, used between vairous classes.                             */
+
+class ArgvParser
+{
+    private array $argv;
+    private array $used;
+
+    public function __construct( array $argv )
+    {
+        $this->argv = $argv;
+        $this->used = [];
+        foreach( $argv as $_ )
+            $this->used[] = false;
+    }
+
+    public function at( int $pos ) : string
+    {
+        $this->used[ $pos ] = true;
+        return $this->argv[ $pos ];
+    }
+
+    public function consume( string $equals = null , string $prefix = null , int $position = -1 ) : string|null
+    {
+        $args = $this->argv;
+        $limit = count( $args );
+
+        for ( $pos = 0 ; $pos < $limit ; $pos++ )
+        {
+            $arg = $args[ $pos ];
```

Comment:
```suggestion
        foreach ( $args as $pos => $arg )
        {
```

Reply:
Nice one. The original array need to be filtered and keys reordered on constructor to garantee `$pos` is `int`. I will do it latter.